src/lib/ndpi_main.c

```diff
@@ -8375,6 +8428,31 @@ static ndpi_protocol ndpi_internal_detection_process_packet(struct ndpi_detectio
 
   if(flow->num_processed_pkts == 1) {
     /* first packet of this flow to be analyzed */
+        
+/*ToDo: First check in fpc cache, if found update the ndpi_flow_struct via fpc_update()
+        else update via below method from flow->guessed_protocol_id_by_ip, NDPI_FPC_CONFIDENCE_IP) and return(ret)
+**use --> ndpi_lru_find_cache() metod and value return from it(dummy) as protocol
+update the flow->fpc_app structue as results
+
+*/
+
+/* For each flow (at the very first packet) lookup into the cache using "SRC IP - DST IP" as key and, if a match is found, save the protocol id into ndpi_flow_struct structure as a "fpc result" */
+//printf("\nApp proto from ret-->%u\n",ret.app_protocol);
+
+/* Check some fpc cache */
+  if(ret.app_protocol == NDPI_PROTOCOL_UNKNOWN &&
+     ndpi_str->fpc_dns_cache &&
+     ndpi_lru_find_cache(ndpi_str->fpc_dns_cache, make_fpc_dns_cache_key(flow),
+			 &fpc_dns_cached_proto, 0 /* Don't remove it as it can be used for other connections */,
+			 ndpi_get_current_time(flow))) {
+    NDPI_LOG_DBG(ndpi_str,"\nFound from FPC cache...%u\n",fpc_dns_cached_proto);
+    ndpi_set_detected_protocol(ndpi_str, flow, fpc_dns_cached_proto, NDPI_PROTOCOL_UNKNOWN, NDPI_FPC_CONFIDENCE_DNS);
```

Comment:
The goal of this work is to set the FPC classification, not the "general" one! So no calls to `ndpi_set_detected_protocol()`!!

Reply:
I will arrange fixes as advice.